src/ScType.ts

```diff
@@ -137,127 +272,329 @@ export class ScType {
-    sc_type_node | sc_type_const | sc_type_node_material
-  );
-
-  static readonly NodeVarStruct = new ScType(
-    sc_type_node | sc_type_var | sc_type_node_struct
-  );
-  static readonly NodeVarTuple = new ScType(
-    sc_type_node | sc_type_var | sc_type_node_tuple
-  );
-  static readonly NodeVarRole = new ScType(
-    sc_type_node | sc_type_var | sc_type_node_role
-  );
-  static readonly NodeVarNoRole = new ScType(
-    sc_type_node | sc_type_var | sc_type_node_norole
-  );
-  static readonly NodeVarClass = new ScType(
-    sc_type_node | sc_type_var | sc_type_node_class
-  );
-  static readonly NodeVarAbstract = new ScType(
-    sc_type_node | sc_type_var | sc_type_node_abstract
-  );
-  static readonly NodeVarMaterial = new ScType(
-    sc_type_node | sc_type_var | sc_type_node_material
-  );
+  static readonly ConstNode = new ScType(sc_type_const | sc_type_node);
+  static readonly VarNode = new ScType(sc_type_var | sc_type_node);
+  static readonly ConstConnector = new ScType(sc_type_const | sc_type_connector);
+  static readonly VarConnector = new ScType(sc_type_var | sc_type_connector);
+  static readonly ConstCommonEdge = new ScType(sc_type_const | sc_type_common_edge);
+  static readonly VarCommonEdge = new ScType(sc_type_var | sc_type_common_edge);
+  static readonly ConstArc = new ScType(sc_type_const | sc_type_arc);
+  static readonly VarArc = new ScType(sc_type_var | sc_type_arc);
+  static readonly ConstCommonArc = new ScType(sc_type_const | sc_type_common_arc);
+  static readonly VarCommonArc = new ScType(sc_type_var | sc_type_common_arc);
+  static readonly ConstMembershipArc = new ScType(sc_type_const | sc_type_membership_arc);
+  static readonly VarMembershipArc = new ScType(sc_type_var | sc_type_membership_arc);
+
+  // permanency
+  static readonly PermArc = new ScType(sc_type_perm_arc);
+  static readonly TempArc = new ScType(sc_type_temp_arc);
+
+  static readonly ConstPermArc = new ScType(sc_type_const | sc_type_perm_arc);
+  static readonly VarPermArc = new ScType(sc_type_var | sc_type_perm_arc);
+  static readonly ConstTempArc = new ScType(sc_type_const | sc_type_temp_arc);
+  static readonly VarTempArc = new ScType(sc_type_var | sc_type_temp_arc);
+
+  // actuality
+  static readonly ActualTempArc = new ScType(sc_type_actual_arc | sc_type_temp_arc);
+  static readonly InactualTempArc = new ScType(sc_type_inactual_arc | sc_type_temp_arc);
+
+  static readonly ConstActualTempArc = new ScType(sc_type_const | sc_type_actual_arc | sc_type_temp_arc);
+  static readonly VarActualTempArc = new ScType(sc_type_var | sc_type_actual_arc | sc_type_temp_arc);
+  static readonly ConstInactualTempArc = new ScType(sc_type_const | sc_type_inactual_arc | sc_type_temp_arc);
+  static readonly VarInactualTempArc = new ScType(sc_type_var | sc_type_inactual_arc | sc_type_temp_arc);
+
+  // positivity
+  static readonly PosArc = new ScType(sc_type_pos_arc);
+  static readonly NegArc = new ScType(sc_type_neg_arc);
+
+  // fuzzy sc-arc
```

Comment:
```suggestion
  // fuzziness
```

Reply:
and place it after // positive sc-arcs section to follow common style when new subtype is introduced and then it is combined with all possible types

Reply:
and place it before line that starts with '// fuzzy sc-arcs'

Reply:
fixed